__tests__/unit-tests/schema/mutation/addStep.mutation.spec.ts

```diff
@@ -0,0 +1,247 @@
+        'common.errors.permissionNotGranted'
+      );
+    });
+  });
+
+  describe('Argument Validation', () => {
+    it('should throw an error if workflow ID is not provided or invalid', async () => {
+      const newArgs = { ...args, workflow: null };
+      const result = addStep.resolve(null, newArgs, context);
+      await expect(result).rejects.toThrow(GraphQLError);
+      expect(context.i18next.t).toHaveBeenCalledWith(
+        'mutations.step.add.errors.invalidArguments'
+      );
+    });
+
+    it('should throw an error if type is not in the valid content types', async () => {
+      const newArgs = { ...args, type: 'invalid' } as unknown as AddStepArgs;
+      const result = addStep.resolve(null, newArgs, context);
+      await expect(result).rejects.toThrow(GraphQLError);
+      expect(context.i18next.t).toHaveBeenCalledWith(
+        'mutations.step.add.errors.invalidArguments'
+      );
+    });
+  });
+
+  describe('Data Retrieval Logic', () => {
+    it('should throw an error if the page linked to the workflow does not exist', async () => {
+      const newArgs = { ...args, workflow: new Types.ObjectId().toHexString() };
+      const result = addStep.resolve(null, newArgs, context);
+      await expect(result).rejects.toThrow(GraphQLError);
+      expect(context.i18next.t).toHaveBeenCalledWith(
+        'common.errors.dataNotFound'
+      );
+    });
+
+    it('should throw an error if the application linked to the page does not exist', async () => {
+      jest.spyOn(Page, 'findOne').mockResolvedValueOnce(null);
+      const result = addStep.resolve(null, args, context);
+      await expect(result).rejects.toThrow(GraphQLError);
+      expect(context.i18next.t).toHaveBeenCalledWith(
+        'common.errors.dataNotFound'
+      );
+    });
+
+    it('should throw an error if the workflow associated with the workflow ID does not exist', async () => {
+      jest.spyOn(Workflow, 'findById').mockResolvedValueOnce(null);
+      const result = addStep.resolve(null, args, context);
+      await expect(result).rejects.toThrow(GraphQLError);
+      expect(context.i18next.t).toHaveBeenCalledWith(
+        'common.errors.dataNotFound'
+      );
+    });
+
+    it('should throw an error if the form linked to the content ID does not exist (when type is not "dashboard")', async () => {
+      const newArgs = {
+        ...args,
+        type: 'form',
+        content: new Types.ObjectId().toHexString(),
+      };
+      const result = addStep.resolve(null, newArgs, context);
+      await expect(result).rejects.toThrow(GraphQLError);
+      expect(context.i18next.t).toHaveBeenCalledWith(
+        'common.errors.dataNotFound'
+      );
+    });
+  });
+
+  describe('Step Creation Logic', () => {
+    it('should create a linked Dashboard if type is "dashboard"', async () => {
+      const newArgs = { ...args, type: 'dashboard' } as unknown as AddStepArgs;
+      jest.spyOn(Role, 'find').mockResolvedValueOnce([]);
```

Comment:
@TaiKamilla 

I fixed all the tests by using MockResolvedValueOnce instead of MockResolvedValue

it prevents the mock to remain between tests